builtin/providers/aws/data_source_aws_availability_zone.go

```diff
@@ -0,0 +1,83 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsAvailabilityZone() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsAvailabilityZoneRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"name_suffix": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"state": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsAvailabilityZoneRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	req := &ec2.DescribeAvailabilityZonesInput{}
+
+	if name := d.Get("name"); name != "" {
+		req.ZoneNames = []*string{aws.String(name.(string))}
+	}
+
+	req.Filters = buildEC2FilterList(d, map[string]string{
+		"state": "state",
+	})
+
+	log.Printf("[DEBUG] DescribeAvailabilityZones %#v\n", req)
```

Comment:
General _going forward_ nit, all* of the AWS structs implement GoStringer, so you can use `%s` to output them to string instead of `%#v`


\* probably